Libs/Optimize/ParticleSystem/VtkMeshWrapper.h

```diff
@@ -24,7 +24,7 @@ class VtkMeshWrapper : public MeshWrapper {
 
   double ComputeDistance(PointType pointa, PointType pointb) const override;
 
-  PointType GeodesicWalk(PointType p, int idx, VectorType vector) const override;
+  PointType GeodesicWalk(PointType p, int idx, VectorType vector, const std::shared_ptr<itk::Constraints> &constraint) const override;
```

Comment:
Do the constraints change during execution?  Do they need to be passed as a parameter for every walk/particle update?

Also, in my opinion a const reference to a shared_ptr is almost always wrong.  See here for explanation:

https://medium.com/@vgasparyan1995/pass-by-value-vs-pass-by-reference-to-const-c-f8944171e3ce

Reply:
They do not change. You're right that it should not be const reference. I still think passing the pointer is the best choice here, since the point of the constraint class is to only have one place that stores the constraints.

Reply:
What I mean is that it appears that it's passing in the same pointer every time (per domain).  What if we change MeshWrapper to have a SetConstraints function that takes and holds this pointer and then internally uses it for the geodesic walk?